### PR TITLE
DDF-04498 Migrate metacard-archive to react, loading bug fix

### PR DESCRIPTION
**Pull request by andrewzimmer**
#### What does this PR do?
Fixes bug in metacard-archive tab that prevented the loading view from displaying after the archive button is clicked.
Also adds an error message to restore
#### Who is reviewing it? 
@adimka 
@Bdthomson 
@hayleynorton 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@djblue

#### How should this be tested?
- Ingest some data
- in the inspector, archive the result
- verify that the loading view displays
- verify an error message appears when attempting to restore
#### What are the relevant tickets?
For Jira:
[](https://codice.atlassian.net/browse/)

For GH Issues:
Fixes: #4498 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/loading-companion/loading-companion.tsx [hunk 1]
```diff
@@ -31,9 +31,9 @@ export default class LoadingCompanionContainer extends React.Component<
   }
   ref = React.createRef()
   componentDidUpdate() {
-    if (this.props.loading === false) {
-      LoadingCompanionView.stopLoadingElement(this.ref.current)
-    }
+    this.props.loading === false
```
Comment (hayleynorton) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/container/loading-companion/loading-companion.tsx]: ✏️ 
```suggestion
    !this.props.loading
```

Comment (Bdthomson) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/loading-companion/loading-companion.tsx]: I'd go even further and say you should just swap the ternary and leave as `this.props.loading`

**Comment by andrewzimmer** [comment]
build now

**Comment by Bdthomson** [comment]
Before
image

After
image

✅ Hero Successful

- [x] Verified that the component looks correct in all theme sizes and also respects custom theme coloring.
- [x] Verified that the restore functionality, while still broken on the backend, now throws an error instead of being silent.

**Comment by andrewzimmer** [comment]
build now

**Comment by andrewzimmer** [comment]
build now

**Pull request merged**